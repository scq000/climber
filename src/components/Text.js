import React from 'react';
import {Text as NativeText, StyleSheet} from 'react-native';

const Text = (props) => (
  <NativeText {...props} style={[styles.text, props.style]} />
);

const styles = StyleSheet.create({
  text: {
    fontFamily: 'Helvetica',
  },
});

export default Text;
