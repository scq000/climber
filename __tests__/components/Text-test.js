import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Text from '../../src/components/Text';

describe('Text', () => {
  it('should render empty text correctly', () => {
    const snap = renderer.create(<Text />).toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('should render not empty text correctly', () => {
    const snap = renderer.create(<Text>test</Text>).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
