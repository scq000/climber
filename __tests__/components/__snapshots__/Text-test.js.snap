// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render empty text correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Helvetica",
      },
      undefined,
    ]
  }
/>
`;

exports[`Text should render not empty text correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Helvetica",
      },
      undefined,
    ]
  }
>
  test
</Text>
`;
